onnx/defs/reduction/defs.cc

```diff
@@ -6,6 +6,44 @@
 #include "onnx/defs/schema.h"
 
 namespace ONNX_NAMESPACE {
+void reduceShapeInference(InferenceContext& ctx) {
+  if (!hasNInputShapes(ctx, 1)) {
+    return;
+  }
+
+  int64_t keep_dims = 1;
+  auto attr_proto = ctx.getAttribute("keepdims");
+  if (attr_proto) {
+    keep_dims = attr_proto->i();
+  }
+  auto& input_shape = ctx.getInputType(0)->tensor_type().shape();
+  int64_t input_ndim = input_shape.dim_size();
+  auto output_shape =
+      ctx.getOutputType(0)->mutable_tensor_type()->mutable_shape();
+  std::vector<int64_t> axes;
+  auto axes_proto = ctx.getAttribute("axes");
+  if (axes_proto)
+    axes.assign(axes_proto->ints().begin(), axes_proto->ints().end());
+
+  for (size_t i = 0; i < axes.size(); ++i) {
+    if (axes[i] < 0)
+      axes[i] += input_ndim;
+  }
+  // do we need handle negative axis?
+  for (int i = 0; i < input_ndim; ++i) {
+    // axes empty means reduce all dim
+    if (!axes.empty() && std::find(axes.begin(), axes.end(), i) == axes.end()) {
```

Comment:
Is `axes` required to be sorted?

Reply:
it's not needed based on current design and implementation.